src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
+    private Element<T> topOfStack;
+    private int size;
+
+    private static class Element<T> {
+        private Element<T> prev;
+        private T value;
+
+        public Element(Element<T> prev, T value) {
+            this.prev = prev;
+            this.value = value;
+        }
+    }
+
     public void push(T value) {
 
+        if (size == 0) {
```

Comment:
The check 'if (size == 0)' is correct for determining if the stack is empty before pushing the first element. No changes needed here.